lib/puppetserver/ca/action/generate.rb

```diff
@@ -103,27 +109,60 @@ def run(input)
 
           # Load, resolve, and validate puppet config settings
           settings_overrides = {}
-          # Since puppet expects the key to be called 'dns_alt_names', we need to use that here
-          # to ensure that the overriding works correctly.
-          settings_overrides[:dns_alt_names] = input['subject-alt-names'] unless input['subject-alt-names'].empty?
           puppet = Config::Puppet.new(config_path)
           puppet.load(settings_overrides)
           return 1 if CliParsing.handle_errors(@logger, puppet.errors)
 
+          # We don't want generate to respect the alt names setting, since it is usually
+          # used to generate certs for other nodes
+          alt_names = Puppetserver::Ca::Utils::Config.munge_alt_names(input['subject-alt-names'])
+          if !alt_names.empty?
+            alt_names += ",DNS:#{puppet.settings[:certname]}"
```

Comment:
If someone gave us an explicit list of alt names, then also add the certname as an alt name? I'm not grokking the purpose here.

Reply:
Apparently if SANs are present, the CN is ignored, see https://stackoverflow.com/questions/5935369/ssl-how-do-common-names-cn-and-subject-alternative-names-san-work-together. My testing bore this out.

I'll switch the order here so that the certname is added before the munge, so it is deduplicated.